examples/platforms/simulation/radio.c

```diff
@@ -822,6 +852,7 @@ void radioSendAck(void)
         otMacFrameGenerateImmAck(&sReceiveFrame, sReceiveFrame.mInfo.mRxInfo.mAckedWithFramePending, &sAckFrame);
     }
 
+    otPlatRadioTxAckStarted(aInstance, sAckFrame.mPsdu, sAckFrame.mLength);
```

Comment:
If the IE is filled in callback, you might need to move `radioProcessTransmitSecurity` in line 847 after `otPlatRadioTxAckStarted`. Otherwise, the MIC calculation will be using the old CSL IE.